modules/metastorage/src/main/java/org/apache/ignite/internal/metastorage/impl/MetaStorageCompactionTrigger.java

```diff
@@ -319,4 +322,33 @@ private void cancelLastScheduledFutureBusy() {
             lock.unlock();
         }
     }
+
+    private void startCompactionOnRecoveryAsync() {
+        assert metaStorageManager.recoveryFinishedFuture().isDone();
+
+        long latestCompactionRevisionLocally = metaStorageManager.getCompactionRevisionLocally();
+
+        if (latestCompactionRevisionLocally != -1) {
+            runAsync(() -> inBusyLockSafe(busyLock, () -> storage.compact(latestCompactionRevisionLocally)), compactionExecutor)
```

Comment:
Is there a guarantee this compaction will be finished before subsequent compactions will be started? If it's because `compactionExecutor` is single-threaded, could you please write a comment about this?

Reply:
There is nothing wrong with changing to a pool with several threads.

The new compaction will first update the compaction revision in memory and then begin its execution. Two compactions will not interfere with each other since they write in batches on which we take a lock for recording.